File: Assignment4/VB.NET/Biavariante/Biavariante/ElementoDistribuzione.vb

Imports System.Runtime.InteropServices

Public Class ElementoDisribuzione
    Private name As String
    Private variabili As Dictionary(Of String, Tuple(Of Object, Type))

    Public Sub New(ByVal nome As String)
        Me.name = nome
        Me.variabili = New Dictionary(Of String, Tuple(Of Object, Type))()
    End Sub

    Public Sub setVariable(ByVal name As String, ByVal s As Tuple(Of Object, Type))
        Me.variabili.Add(name, s)
    End Sub

    Public Function getVariable(ByVal name As String, <Out> ByRef ret As Tuple(Of Object, Type)) As Boolean
        If Me.variabili.TryGetValue(name, ret) Then
            Return True
        Else
            Return False
        End If
    End Function

    Public Function getVariabili() As Dictionary(Of String, Tuple(Of Object, Type))
        Return Me.variabili
    End Function
End Class

File: Assignment7/C#/BivarianteGraphics week5/WindowsFormsApp1/ElementoDistribuzione.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsFormsApp1
{
    class ElementoDisribuzione
    {
        private String name;
        private Dictionary<String, Tuple<Object,Type>> variabili;
        public ElementoDisribuzione(String nome)
        {
            this.name = nome;
            this.variabili = new Dictionary<string, Tuple<Object, Type>>();

        }

        public void setVariable(String name, Tuple<Object, Type>s)
        {
            this.variabili.Add(name,s);
        }

        public bool getVariable(String name, out Tuple<Object, Type> ret)
        {
            if (this.variabili.TryGetValue(name, out ret)) return true;
            else return false;
        }
        public Dictionary<String, Tuple<Object, Type>> getVariabili()
        {
            return this.variabili;
        }

    }

}
